app/scripts/migrations/115.ts

```diff
@@ -0,0 +1,82 @@
+import { cloneDeep, isObject } from 'lodash';
```

Comment:
I'm just leaving a comment instead of a full review because this is the first time I've look at migrations and deprecations code before.

Does `incomingTransactionsPreferences` need to be updated at all? (I just noticed `LINEA_GOERLI` was added to it in migration 096 when I was trying to understand how migrations and migrations tests work.)

Reply:
hey @davidmurdoch, for deprecating linea-goerli we wanted to switch users to linea-sepolia and remove "linea-goerli" from the network list options and let users keep using it if they wish to but show them deprecation warning whenever they switch to it.
Don't think we need to update incomingTransactionsPreferences, 
Here is the scenario that will happen regarding incoming transactions:
1- User is on linea goerli and he has received transactions (he can see them on activity tab)
2- This new migration runs and the user finds himself switched to linea Sepolia, and the linea goerli network is no longer displayed as an option in the list of networks in MM
3- User manually adds linea goerli network and switches to it (he will see deprecation warning)
4- User goes to activity tab and he can still see transactions he received before.

Reply:
hey @davidmurdoch ! thanks for checking this 🙏  
I don't think we need to update the `state.PreferencesController.incomingTransactionsPreferences`;
For deprecating linea-goerli; we wanted to switch users to linea-sepolia, remove "linea-goerli" from the network list options and let users keep using it if they wish to but show them deprecation warning whenever they switch to it.

When this migration runs, the `state.PreferencesController.incomingTransactionsPreferences`  value will be whatever values the user has already set. 

If for example the user had the toggles in Settings => Security and privacy => transactions `ON` for linea-goerli and linea-Sepolia. Then this migration runs, we will have this object :
```
state.PreferencesController.incomingTransactionsPreferences = {
    ...
    "0xe704": true, // linea-goerli
    "0xe705": true, //linea-sepolia
}
```

And users will:
1 - Find themselves on linea-sepolia
2- They wont see linea-goerli in the network list
3- If they decide to manually add it, they will see deprecation warning and they use it knowing its deprecated.